service/cloudconfigv2/cloud_config.go

```diff
@@ -51,32 +53,26 @@ func New(config Config) (*CloudConfig, error) {
 
 // NewMasterTemplate generates a new worker cloud config template and returns it
 // as a base64 encoded string.
-func (c *CloudConfig) NewMasterTemplate(customObject v1alpha1.KVMConfig, certs certificatetpr.AssetsBundle, node v1alpha1.ClusterNode) (string, error) {
+func (c *CloudConfig) NewMasterTemplate(customObject v1alpha1.KVMConfig, certs certificatetpr.AssetsBundle, node v1alpha1.ClusterNode, keys map[randomkeytpr.Key][]byte) (string, error) {
 	var err error
-
-	var params k8scloudconfig.Params
-	{
-		params.Cluster = customObject.Spec.Cluster
-		params.Extension = &masterExtension{
-			certs: certs,
-		}
-		params.Node = node
-	}
-
-	var newCloudConfig *k8scloudconfig.CloudConfig
-	{
-		newCloudConfig, err = k8scloudconfig.NewCloudConfig(k8scloudconfig.MasterTemplate, params)
+	var template string
+	switch customObject.Spec.Cluster.Version {
```

Comment:
This does not work. The cluster version field was actually removed. Ross added it back because of some details in AWS. Versioning has to happen via version bundles and resource routing. IMO the cloud config package should stay as it is. We should rather introduce another one which does the magic, which implies that another configmap resource packages has to be introduced then as well. Otherwise we are either not able to version or screw existing clusters like before.

Reply:
Ok I would need a deeper explanatoin on version bundle and resource routing so

Reply:
Check 

- https://github.com/giantswarm/aws-operator/blob/master/service/resource_router.go
- https://github.com/giantswarm/aws-operator/blob/master/service/version_bundles.go

Note that the resource routing is not yet implemented so this should happen as a first step in a separate PR. 

Another PR should then be to copy the cloudconfig package and the configmap resource package. 

Another PR should then be to add changes to the copied packages accordingly. 

Another PR should then be to add the new resource to the resource router.